app/components/viewer/ReplayCars.tsx

```diff
@@ -0,0 +1,254 @@
+        child.material = matClone;
+    });
+
+    useFrame((state, delta) => {
+        if (!mesh || !mesh.current) {
+            return;
+        }
+
+        if (mesh && mesh.current && mesh.current) {
+            sampleIndex = 0;
+            while (sampleIndex + 1 < replay.samples.length
+                    && replay.samples[sampleIndex].currentRaceTime < timeLineGlobal.currentRaceTime) {
+                sampleIndex++;
+            }
+            CurrentSampleRef.current = replay.samples[sampleIndex];
+
+            const distToCamera = replay.samples[sampleIndex].position.distanceTo(camera.position);
+
+            if (hovered) {
+                (meshTxt.current.children[0] as any).text = `
+                ${replay.playerName}\n
+                ${getRaceTimeStr(replay.endRaceTime)}\n
+                cameraMode: ${cameraMode}\n
+                (click to focus/unfocus)`;
+            } else {
+                (meshTxt.current.children[0] as any).text = '';
+            }
+
+            meshTxt.current.scale.set(distToCamera / 400, distToCamera / 400, distToCamera / 400);
+            meshTxt.current.rotation.set(
+                camera.rotation.x,
+                camera.rotation.y,
+                camera.rotation.z,
+            );
+
+            mesh.current.position.lerp(replay.samples[sampleIndex].position, 0.4);
+            stadiumCarMesh.current.rotation.setFromQuaternion(
+                vecToQuat(replay.samples[sampleIndex].dir, replay.samples[sampleIndex].up),
+            );
+
+            stadiumCarMesh.current.children[2].rotation.y = replay.samples[sampleIndex].wheelAngle;
+            stadiumCarMesh.current.children[4].rotation.y = replay.samples[sampleIndex].wheelAngle;
+
+            velocityRef.current.rotation.setFromQuaternion(
+                vecToQuat(replay.samples[sampleIndex].dir, replay.samples[sampleIndex].up),
+            );
+
+            const camPos = replay.samples[sampleIndex].velocity.clone().negate().divideScalar(3);
+            camPos.x += 4;
+            camPos.y += 5;
+            velocityRef.current.position.lerp(camPos, 0.4);
+
+            if (timeLineGlobal.followedReplay != null && timeLineGlobal.followedReplay._id === replay._id) {
+                if (orbitControlsRef && orbitControlsRef.current) {
+                    orbitControlsRef.current.target.lerp(replay.samples[sampleIndex].position, 0.4);
+                    if (cameraMode === '1') {
+                        const camWorldPos: THREE.Vector3 = new THREE.Vector3();
+                        velocityRef.current.getWorldPosition(camWorldPos);
+                        camera.position.lerp(camWorldPos, 0.3);
+                    }
+                }
+            }
+        }
+    });
+
+    const onPointerLeave = (e: MouseEvent) => {
+        hovered = false;
+    };
+
+    const onPointerMove = (e: MouseEvent) => {
+        hovered = true;
+    };
+
+    const onClick = () => {
+        if (Date.now() - prevOnClick > 100) {
```

Comment:
Is this timeout of 100ms really necessary?

Reply:
It registers multiple events per single click, the timeout is a workaround to prevent simultaneous selection => deselection